repos/spack_repo/builtin/packages/erf/package.py

```diff
@@ -0,0 +1,174 @@
+
+
+class Erf(CMakePackage, CudaPackage, ROCmPackage):
+    """ERF solves the compressible Navier-Stokes on a Arakawa C-grid
+    for large-scale weather modeling.
+    """
+
+    homepage = "https://erf.readthedocs.io/en/latest/index.html"
+    url = "https://github.com/erf-model/ERF/archive/refs/tags/25.08.tar.gz"
+    git = "https://github.com/erf-model/ERF.git"
+    test_requires_compiler = True
+
+    maintainers("larenspear")
+
+    license("BSD-3-Clause", checked_by="larenspear")
+
+    version("25.10", tag="25.10", submodules=True)
+    version("25.08", tag="25.08", submodules=True)
+    version("25.07", tag="25.08", submodules=True)
+    version("25.06", tag="25.08", submodules=True)
+    version("25.05", tag="25.08", submodules=True)
+    version("25.04", tag="25.08", submodules=True)
+    version("25.03", tag="25.08", submodules=True)
+    version("25.01", tag="25.08", submodules=True)
+    version("24.11", tag="25.08", submodules=True)
+    version("24.10", tag="25.08", submodules=True)
+    version("24.09", tag="25.08", submodules=True)
+    version("24.08", tag="25.08", submodules=True)
+    version("24.06", tag="25.08", submodules=True)
+    version("24.05", tag="25.08", submodules=True)
+    version("24.04", tag="25.08", submodules=True)
+    version("24.03", tag="25.08", submodules=True)
+    version("24.02", tag="25.08", submodules=True)
+    version("24.01", tag="25.08", submodules=True)
+    version("23.12", tag="25.08", submodules=True)
+    version("23.11", tag="25.08", submodules=True)
+    version("23.10", tag="25.08", submodules=True)
+
+    variant("mpi", default=False, description="Enable MPI support")
+    variant("openmp", default=False, description="Enable OpenMP support")
+    variant("hip", default=False, description="Enable HIP/rocm support")
+    variant("sycl", default=False, description="Enable SYCL support")
+    variant("netcdf", default=False, description="Enable NetCDF support")
+    variant("particles", default=False, description="Enable particle support")
+    variant("multiblock", default=False, description="Enable multiblock support")
+    variant("radiation", default=False, description="Enable radiation support")
+    variant("tests", default=False, description="Enable tests")
+    variant("fcompare", default=False, description="Enable fcompare")
+    variant("fft", default=False, description="Enable FFT support")
+    variant("debug", default=False, description="Enable debug build")
+
+    depends_on("cmake@3.20:", type="build")
+    depends_on("git", type="build")
+    depends_on("amrex", type=("build", "link", "run"))
+    for v in ("mpi", "openmp", "cuda", "sycl", "particles"):
+        depends_on(f"amrex+{v}", when=f"+{v}")
+        depends_on(f"amrex~{v}", when=f"~{v}")
+    for sm in CudaPackage.cuda_arch_values:
+        depends_on(f"amrex+cuda cuda_arch={sm}", when=f"+cuda cuda_arch={sm}")
+    depends_on("amrex+rocm", when="+hip")
+    depends_on("amrex~rocm", when="~hip")
```

Comment:
```suggestion
    depends_on("amrex", type=("build", "link", "run"))
    for v in ("mpi", "openmp", "cuda", "sycl", "particles"):
        depends_on(f"amrex+{v}", when=f"+{v}")
        depends_on(f"amrex~{v}", when=f"~{v}")
    for sm in CudaPackage.cuda_arch_values:
        depends_on(f"amrex+cuda cuda_arch={sm}", when=f"+cuda cuda_arch={sm}")
    depends_on("amrex+rocm", when="+hip")
    depends_on("amrex~rocm", when="~hip")
```
You need to either repeat the type arguments for all of the dependencies they apply to, or use a `with default_args(type=("build", "link", "run")):` context manager around them.